File: SINGLEOP1.py
# cook your dish here
count=0
for i in range(int(input())):
    n=int(input())
    s=(input())
    dec=int(s,2)
    #print(dec)
    #sqr=int(m.sqrt(dec))
    #print("sqr = ",sqr)
    for j in range(len(s)):
        if(s[j]=='1'):
            count+=1
            #print("count = ",count)
        else:
            break 
            
    print(count)
    count=0
    

File: SINGLEOP1.cpp
#include <iostream>
using namespace std;

int main() {
    int t;
    cin>>t;
    while(t--)
    {
        int n,i,c=0;
        cin>>n;
        string s;
        cin>>s;
        for(i=0;i<n;i++)
        {
            if(s[i]=='1')
                c++;
            else
                break;
        }
        cout<<c<<endl;
        
    }
	// your code goes here
	return 0;
}